src/WebJobs.Script.Extensibility/ScriptBindingContext.cs

```diff
@@ -107,12 +114,32 @@ public ScriptBindingContext(JObject bindingMetadata)
         public TValue GetMetadataValue<TValue>(string name, TValue defaultValue = default(TValue))
         {
             JToken value = null;
-            if (Metadata.TryGetValue(name, StringComparison.OrdinalIgnoreCase, out value))
+            try
             {
-                return value.Value<TValue>();
+                if (Metadata.TryGetValue(name, StringComparison.OrdinalIgnoreCase, out value))
+                {
+                    return value.Value<TValue>();
+                }
+            }
+            catch (FormatException e)
+            {
+                throw new FormatException($"Invalid value specified for binding property '{name}' of type {PrettyTypeName(typeof(TValue))}.", e);
             }
 
             return defaultValue;
         }
+
+        private static string PrettyTypeName(Type t)
```

Comment:
I don't think having the namespace there would impact readability. It would be confusing if we don't have the namespace and we end up dealing with enumerations/types that have the same name, as it wouldn't be immediately clear what type was being used.